File: robocode.tests.robots/src/main/java/tested/robots/BadFirePower.java
/**
 * Copyright (c) 2001-2019 Mathew A. Nelson and Robocode contributors
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * https://robocode.sourceforge.io/license/epl-v10.html
 */
package tested.robots;


import robocode.AdvancedRobot;
import robocode.Bullet;


/**
 * From Bug 2410856
 * @author Pavel Savara (original)
 */
public class BadFirePower extends AdvancedRobot {

	public void run() {
		while (true) {
			Bullet bullet = setFireBullet(getTime() - 12.0);

			if (bullet != null) {
				out.println(getTime() + " Bullet power: " + bullet.getPower());
			} else {
				out.println(getTime() + " No bullet");
			}
			execute();
		}
	}

}

File: plugins/dotnet/robocode.dotnet.robotscs/src/tested/robotscs/BadFirePower.cs
/**
 * Copyright (c) 2001-2019 Mathew A. Nelson and Robocode contributors
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * https://robocode.sourceforge.io/license/epl-v10.html
 */
using Robocode;

namespace tested.robotscs
{
    /**
     * @author Pavel Savara (original)
     */

    public class BadFirePower : AdvancedRobot
    {
        public override void Run()
        {
            while (true)
            {
                Bullet bullet = SetFireBullet(Time - 12.0);

                if (bullet != null)
                {
                    Out.WriteLine(Time + " Bullet power: " + bullet.Power);
                }
                else
                {
                    Out.WriteLine(Time + " No bullet");
                }
                Execute();
            }
        }
    }
}